src/contents/all/components/knot/KnotInput/KnotInput.tsx

```diff
@@ -2,36 +2,37 @@ import React, { useState, useRef, useEffect } from 'react';
 import { Timeline } from "antd";
 import RichTextEditor from 'react-rte/lib/RichTextEditor';
 
-import { User, Knot } from '../../../global/types';
+import { User, Knot, Log } from '../../../global/types';
 
 import KeyboardShortcut from '../../common/KeyboardShortcut/KeyboardShortcut'
 
 import "../Knot/Knot.scss"
 import "./KnotInput.scss";
 
 interface Props {
+  knot?: string
   createKnot: (bodyHtml: string, bodyText: string) => void
-  deleteKnot: any
-  editKnot: any
-  hasKnots: boolean
+  editKnot?: (bodyHtml: string, bodyText: string) => void
+  hasKnots?: boolean
   show: boolean
 }
 
-const KnotInput = ({createKnot, hasKnots, show}: Props) => {
+const KnotInput = ({knot, createKnot, editKnot, hasKnots, show}: Props) => {
 
   const knotEditor = useRef(null)
 
   const [submit, setSubmit] = useState(false)
   const [showFooter, setShowFooter] = useState(false)
   const [body, setBody] = useState(RichTextEditor.createEmptyValue())
-  const [knot, setKnot] = useState<Knot>(null)
 
   const [loading, setLoading] = useState(false)
-  const [deleteLoading, setDeleteLoading] = useState(false)
 
   useEffect(() => {
     const targetIsRef = knotEditor.hasOwnProperty("current")
     const currentTarget = targetIsRef ? knotEditor.current : knotEditor;
+    if (knot) {
+      setBody(knot)
+    }
```

Comment:
this is pretty pseudocode-y, but my intuition is to have KnotInput take an optional knot param and render that text immediately as body text (so existing knots are editable). probably (k)not doing this right.

Reply:
hahah that pun just made my day. 

Yea I think you're intuition is spot on here. I might try to make it render just the Knot with no editor and then when you click on the Knot, it will render the KnotInput and focus the editor. That way you don't have to render tons of KnotInput for every single Knot and it renders all that functionality dynamically only when necessary.

Thoughts?